src/glossary.rst

```diff
@@ -967,3 +967,117 @@ Glossary
       See also:
 
       * :ref:`Graph Branching`
+
+source workflow
+   A source workflow is a :term:`cylc workflow` that is not located in the
+   :term:`cylc run directory`.
+
+   In order to be :term:`run` source workflows must be
+   :term:`installed <install>` into the :term:`cylc run directory`.
+
+workflow
+cylc workflow
+   A Cylc workflow is a collection of things to run (:term:`tasks`) along with
+   the constraints that govern the order in which they run
+   (:term:`dependencies`) represented in Cylc format in a
+   :cylc:conf:`flow.cylc` file.
+
+   For example here is a Cylc workflow representing the brewing process:
+
+   .. code-block:: cylc
+      :caption: flow.cylc
+
+      [scheduling]
+          cycling mode = integer
+          initial cycle point = 1
+          [[graph]]
+              # repeat this for each batch
+              P1 = """
+                  # the stages of brewing in the order they must occur in
+                  malt => mash => sparge => boil => chill => ferment => rack
+
+                  # must finish the sparge of one batch before
+                  # starting the next one
+                  # sparge[-P1] => mash
+              """
+
+.. TODO - add this after the universal ID work
+
+   workflow id
+      A workflow ID is the name Cylc uses to identify a :term:`workflow` in
+      the :term:`run directory`.
+
+      This ID is used on the command line and in the GUI.
+
+      The ID is the relative path between the :term:`workflows <workflow>`
+      :term:`run directory` and the :term:`cylc run directory`.
+
+      For example, if your workflow is in ``~/cylc-run/foo/bar/run1`` then its ID
+      is ``foo/bar/run1``.
+
+      .. ..note:: For convenience you can omit the ``/run1`` on the command line.
+
+flow
+   A flow is a single logical run of a :term:`workflow` that is done by
+   a :term:`scheduler`.
+
+   A flow can be :term:`played <play>` and :term:`paused <pause>`,
+   :term:`stopped <stop>` and :term:`restarted <restart>`.
+
+   A flow begins at the :term:`start cycle point` and ends at the
+   :term:`stop cycle point`.
+
+   It is possible to run more than one flow in a single :term:`scheduler`.
+
+reflow
+   A reflow is a subsequent logical run of a :term:`workflow` that is done by
+   the same :term:`scheduler` as the original :term:`flow`.
+
+   Reflows are useful when you need to re-wind your :term:`workflow` run to
+   allow it to evolve a new path into the future.
+
+   .. TODO - add this once the CLI example works
+
+      For example, the following workflow runs the task ``a`` every cycle, then
+      either ``b`` or ``c`` depending on the outcome of ``a``:
+
+      .. code-block:: cylc
+
+         [scheduling]
+             cycling mode = integer
+             initial cycle point = 1
+             [[graph]]
+                 P1 = """
+                     b[-P1] | c[-P1] => a
+                     a:succeed => b
+                     a:fail => c
+                 """
```

Comment:
We'll need to start adding `allow implicit tasks = True` to all our example workflows so that they can run out of the box.

Reply:
Ach, dammit, that makes examples so convoluted.

Reply:
or define the tasks in the runtime section?

Reply:
Even worse?